### PR TITLE
Precommit: ensure executable bit is set

### PR DESCRIPTION
**Pull request by jnewb1**
Added two new precommit rules

- if it has a shebang, it should be executable
- if it's set executable and not binary, it should have a shebang

also fixed a bunch of violations of these rules and removed shebangs where it doesn't make sense to have it be executable

### CODE REVIEW

**Review by adeebshihadeh — APPROVED** [review]
Nice, let's bring to the submodules as well

**Comment by sshane** [comment]
You can bump the panda submodule up to before the change this PR and push to master, so the submodule diff isn't so big. It's also a good idea to bump it when we merge new safety, so we can start getting test_models coverage.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,6 +14,10 @@ repos:
     -   id: check-yaml
     -   id: check-merge-conflict
     -   id: check-symlinks
+    -   id: check-executables-have-shebangs
+        exclude: '^(tinygrad_repo)'
```
Comment (adeebshihadeh) [inline .pre-commit-config.yaml:18]: let's make this a global exclude up on top (and only tinygrad for now)

Comment (jnewb1) [reply .pre-commit-config.yaml]: done

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**